snuba/datasets/discover.py

```diff
@@ -111,6 +113,35 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
         source.set_table_source(detected_dataset)
 
 
+class SimpleColumnValidator(QueryProcessor):
+    """
+    Checks that requested columns are present on the Discover schema. Prevents a column
+    that happens to be present on an underlying dataset being inadvertently exposed
+    on Discover.
+    """
+
+    def __init__(self, all_columns: ColumnSet) -> None:
+        self.__columns = all_columns
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        aliases = query.get_aliases()
+        columns = [
+            col for col in query.get_all_referenced_columns() if col not in aliases
+        ]
+        time_group_columns = [
+            "time",
+            "bucketed_end",
+        ]  # TODO: Remove when "time" column is correctly handled
+        for col in columns:
+            if (
+                col
+                and col not in self.__columns
+                and not NESTED_COL_EXPR_RE.match(col)
+                and col not in time_group_columns
+            ):
```

Comment:
Does this work with apdex (and the other function expressions)? 
https://github.com/getsentry/snuba/blob/master/snuba/util.py#L61-L100

Reply:
No, we are relying on `get_all_referenced_columns` here which doesn't extract the column name for function expressions. So, we will be skipping the validation on columns where this syntax is used.